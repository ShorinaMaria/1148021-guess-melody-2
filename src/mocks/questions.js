const questions = [
  {
    type: `genre`,
    genre: `jazz`,
    answers: [
      {
        src: `http://mp3.classic-music.ru/music/vivaldi/01.mp3`,
        genre: `classic`,
      },
      {
        src: `http://dl2.mp3party.net/online/8464242.mp3`,
        genre: `pop`,
      },
      {
        src: `http://dl2.mp3party.net/online/2072.mp3`,
        genre: `jazz`,
      },
      {
        src: `http://dl2.mp3party.net/online/7625333.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `http://mp3.classic-music.ru/music/vivaldi/01.mp3`,
        genre: `classic`,
      },
      {
        src: `http://dl2.mp3party.net/online/914916.mp3`,
        genre: `rock`,
      },
      {
        src: `http://dl2.mp3party.net/online/2072.mp3`,
        genre: `jazz`,
      },
      {
        src: `http://dl2.mp3party.net/online/8468238.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Placebo`,
      src: `http://dl2.mp3party.net/online/8468238.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Placebo`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Scorpions`,
      src: `http://dl2.mp3party.net/online/7625333.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `Scorpions`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
];

export default questions;
