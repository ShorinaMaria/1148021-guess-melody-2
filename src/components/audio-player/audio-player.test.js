import React from 'react';
import questions from '../../mocks/questions';
import AudioPlayer from './audio-player';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

it(`AudioPlayer renders correctly`, () => {
  const audioFile = questions[0].answers[0].src;
  const component = mount(
      <AudioPlayer
        isPlaying={false}
        onPlayButtonClick={()=>{}}
        src={audioFile}
      />
  );

  expect(toJson(component)).toMatchSnapshot();
});
