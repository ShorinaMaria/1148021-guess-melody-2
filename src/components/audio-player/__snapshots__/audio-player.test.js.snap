// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer renders correctly 1`] = `
<AudioPlayer
  isPlaying={false}
  onPlayButtonClick={[Function]}
  src="http://mp3.classic-music.ru/music/vivaldi/01.mp3"
>
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
</AudioPlayer>
`;
